web/pandas/pdeps/00xx-string-dtype.md

```diff
@@ -0,0 +1,247 @@
+   dtype that has massive benefits for our users, both in usability as (for the
+   significant part of the user base that has PyArrow installed) in performance.
+2. In case we eventually transition to use `pd.NA` as the default missing value
+   sentinel, we will need a migration path for _all_ our data types, and thus
+   the challenges around this will not be unique to the string dtype and
+   therefore not a reason to delay this.
+
+### Why not use the existing StringDtype with `pd.NA`?
+
+Wouldn't adding even more variants of the string dtype will make things only more
+confusing? Indeed, this proposal unfortunately introduces more variants of the
+string dtype. However, the reason for this is to ensure the actual default user
+experience is _less_ confusing, and the new string dtype fits better with the
+other default data types.
+
+If the new default string data type would use `pd.NA`, then after some
+operations, a user can easily end up with a DataFrame that mixes columns using
+`NaN` semantics and columns using `NA` semantics (and thus a DataFrame that
+could have columns with two different int64, two different float64, two different
+bool, etc dtypes). This would lead to a very confusing default experience.
+
+With the proposed new variant of the StringDtype, this will ensure that for the
+_default_ experience, a user will only see only 1 kind of integer dtype, only
+kind of 1 bool dtype, etc. For now, a user should only get columns with an
+`ArrowDtype` and/or using `pd.NA` when explicitly opting into this.
+
+## Backward compatibility
+
+The most visible backwards incompatible change will be that columns with string
+data will no longer have an `object` dtype. Therefore, code that assumes
+`object` dtype (such as `ser.dtype == object`) will need to be updated.
+
+To allow testing your code in advance, the
+`pd.options.future.infer_string = True` option is available.
+
+Otherwise, the actual string-specific functionality (such as the `.str` accessor
+methods) should all keep working as is. By preserving the current missing value
+semantics, this proposal is also backwards compatible on this aspect.
+
+One other backwards incompatible change is present for early adopters of the
+existing `StringDtype`. In pandas 3.0, calling `pd.StringDtype()` will start
+returning the new default string dtype, while up to now this returned the
+experimental string dtype using `pd.NA` introduced in pandas 1.0. Those users
+will need to start specifying a keyword in the dtype constructor if they want to
+keep using `pd.NA` (but if they just want to have a dedicated string dtype, they
+don't need to change their code).
+
+## Timeline
+
+The future PyArrow-backed string dtype was already made available behind a feature
+flag in pandas 2.1 (by `pd.options.future.infer_string = True`).
+
+Some small enhancements or fixes (or naming changes) might still be needed and
+can be backported to pandas 2.2.x.
```

Comment:
This part of the plan worries me a little.

Maybe it would be better to cut off a 2.3 from 2.2.x.

I think there's a significant proportion of the downloads for 2.2 that aren't on the latest patch release.
I think there's ~ 1/3 of the downloads that are fetching 2.2.0.

Reply:
Also,
it would be good to mention which version of pandas is expected to have  ``infer_string`` be able to infer to the fallback option.

Reply:
a 2.3 release (maybe around the same time as 3.0rc) sounds reasonable. 

If the features/bugfixes added to 2.3 are limited to the string dtype then we shouldn't need many patch releases. We may not need to fix any string dtype related issues that are fixed for 3.0 as these will be behind a flag in 2.3 and so shouldn't break existing code.

On the other hand, as these features are behind a flag, maybe releasing a 2.3 would not gain the field testing we hope for.

And therefore, instead of doing a 2.3, planning for at least a couple of release candidates for 3.0 would better achieve this.

Reply:
@jorisvandenbossche 

Thoughts on this?

Reply:
> Maybe it would be better to cut off a 2.3 from 2.2.x.

Yes, if we still plan to add a deprecation warning and change the naming scheme in `StringDtype`, calling that 2.3.0 sounds as the best option (I had been planning to propose doing a 2.3.0 (from the 2.2.x branch) anyway to bump the warning for CoW from DeprecationWarning to FutureWarning)